src/intervals/conversion_promotion.jl

```diff
@@ -2,14 +2,12 @@
 
 ## Conversion and promotion
 
-## Default conversion is to Interval{Float64}
+## Default conversion to Interval, corresponds to Interval{Float64}
 convert{T<:Real}(::Type{Interval}, x::T) = make_interval(Float64, x)
-@compat convert{T<:Union{Float64,BigFloat,Rational}}(::Type{Interval}, x::T) =
-    make_interval(T, x)
 
-## Conversion to intervals (with rounding) from Integer or Irrational
+## Conversion to specific type intervals
 @compat convert{T<:Union{Float64,BigFloat}, S<:Real}(::Type{Interval{T}}, x::S) =
```

Comment:
I have no problem with the observation about Lint.jl... 

Regarding `AbstractFloat`, do we really want to deal with (or care about) `Float32` or `Float16`? I didn't even think about this possibility before, but actually, `Interval(Float32(1.1))` yields `Interval{Float32}` as result, as it should. However, there is no `make_interval(Float32, x)`, and I am not sure that the rounding modes, or CRlibm care about this. I am in favor of concentrating only in `Interval{T}` with `T=Union{Float64, AbstractFloat, Rational}`, and I am always wondering about the very last one... 

We could certainly use `AbstractFloat` wherever is needed, but at some point the intervals will be promoted to Interval{Float64}, at least.